pkg/controller/stats/aggregator.go

```diff
@@ -369,3 +381,34 @@ func addUp(stats *statsv1alpha1.TrafficStats, inc *statsv1alpha1.TrafficStats) {
 	stats.Packets += inc.Packets
 	stats.Bytes += inc.Bytes
 }
+
+func addRulesUp(ruleStats *[]statsv1alpha1.RuleTrafficStats, ruleSumStats *statsv1alpha1.TrafficStats, inc []statsv1alpha1.RuleTrafficStats) {
+	incMap := make(map[string]*statsv1alpha1.TrafficStats)
+	for i, v := range inc {
+		incMap[v.Name] = &inc[i].TrafficStats
+	}
+	// accumulate incMap traffics stats to the current traffic stats
+	for _, v := range incMap {
+		addUp(ruleSumStats, v)
+	}
+	// accumulate the rule traffic stats if the rule has already 'existed' in the ruleStats
```

Comment:
if the rule already exists in the ruleStats

Reply:
@ceclinux and this one?